packages/compile-solidity/compileWithPragmaAnalysis.js

```diff
@@ -0,0 +1,123 @@
+const CompilerSupplier = require("./compilerSupplier");
+const semver = require("semver");
+const Profiler = require("./profiler");
+const { normalizeOptions } = require("./normalizeOptions");
+const fse = require("fs-extra");
+const { run } = require("./run");
+const OS = require("os");
+
+const getSemverExpression = source => {
+  return source.match(/pragma solidity(.*);/)[1].trim();
+};
+
+const getSemverExpressions = sources => {
+  return sources.map(source => getSemverExpression(source));
+};
+
+// takes an array of versions and an array of semver expressions
+// returns a version of the compiler or undefined if none can be found
+const findNewestSatisfyingVersion = ({ solcReleases, semverExpressions }) => {
+  // releases are ordered from newest to oldest
```

Comment:
I wonder if this ordering will be guaranteed? I don't think we should risk it. The list may even be alphabetically ordered, in which case `0.10.0` will come before `0.2.0`.

Maybe `compilerSupplier` itself should sort the list? (maybe it does already?)

Reply:
This is the order that we get from requesting the list of releases and I'm not sure it is guaranteed. So I think it is probably a good idea to sort it somewhere.